lib/Parse/ParseStmt.cpp

```diff
@@ -854,7 +865,7 @@ StmtResult Parser::ParseCompoundStatement(bool isStmtExpr) {
 ///
 StmtResult Parser::ParseCompoundStatement(bool isStmtExpr,
                                           unsigned ScopeFlags) {
-  assert(Tok.is(tok::l_brace) && "Not a compount stmt!");
+  assert(Tok.is(tok::l_brace) && "Not a compound stmt!");
```

Comment:
I agree that there is a typographical error here.   However, this change is unrelated to parsing checked keywords, so please undo it.  It might make future merges of code harder.